zgw/catalogi-api/src/main/kotlin/com/ritense/catalogiapi/CatalogiApiPlugin.kt

```diff
@@ -121,195 +139,246 @@ class CatalogiApiPlugin(
     }
 
     fun getBesluittypen(zaakTypeUrl: URI): List<Besluittype> {
-        var currentPage = 1
-        var currentResults: Page<Besluittype>?
-        val results = mutableListOf<Besluittype>()
-
-        do {
-            logger.debug { "Getting page of besluittypen, page $currentPage for zaaktype $zaakTypeUrl" }
-            currentResults = client.getBesluittypen(
-                authenticationPluginConfiguration,
-                url,
-                BesluittypeRequest(
-                    zaaktypen = zaakTypeUrl,
-                    page = currentPage++
+        withLoggingContext(CATALOGI_API.BESLUITTYPE to zaakTypeUrl.toString()) {
+            var currentPage = 1
+            var currentResults: Page<Besluittype>?
+            val results = mutableListOf<Besluittype>()
+
+            do {
+                logger.debug { "Getting page $currentPage of besluittypen for zaaktype $zaakTypeUrl" }
+                currentResults = client.getBesluittypen(
+                    authenticationPluginConfiguration,
+                    url,
+                    BesluittypeRequest(
+                        zaaktypen = zaakTypeUrl,
+                        page = currentPage++
+                    )
                 )
-            )
-            results.addAll(currentResults.results)
-        } while (currentResults?.next != null)
+                results.addAll(currentResults.results)
+            } while (currentResults?.next != null)
 
-        return results
+            return results
+        }
     }
 
     fun getEigenschappen(zaakTypeUrl: URI): List<Eigenschap> {
-        return Page.getAll { page ->
-            logger.debug { "Getting page of eigenschappen, page $page for zaaktype $zaakTypeUrl" }
-            client.getEigenschappen(
-                authenticationPluginConfiguration,
-                url,
-                EigenschapRequest(
-                    zaaktype = zaakTypeUrl,
-                    page = page
+        withLoggingContext(CATALOGI_API.EIGENSCHAP to zaakTypeUrl.toString()) {
+            return Page.getAll { page ->
+                logger.debug { "Getting page $page of eigenschappen for zaaktype $zaakTypeUrl" }
+                client.getEigenschappen(
+                    authenticationPluginConfiguration,
+                    url,
+                    EigenschapRequest(
+                        zaaktype = zaakTypeUrl,
+                        page = page
+                    )
                 )
-            )
+            }
         }
     }
 
     fun getBesluittypeByOmschrijving(zaakTypeUrl: URI, omschrijving: String): Besluittype {
-        return getBesluittypen(zaakTypeUrl)
-            .singleOrNull { it.omschrijving.equals(omschrijving, ignoreCase = true) }
-            ?: throw StatustypeNotFoundException("With 'omschrijving': '$omschrijving'")
+        withLoggingContext(
+            CATALOGI_API.BESLUITTYPE to zaakTypeUrl.toString(),
+            CATALOGI_API.BESLUITTYPE to omschrijving
```

Comment:
Both keys are the same, but i think we don't need the omschrijving as MDC variable.